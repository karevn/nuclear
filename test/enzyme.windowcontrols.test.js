import React from 'react';
import { shallow } from 'enzyme';

import FontAwesome from 'react-fontawesome';

import WindowButton from '../app/components/WindowControls/WindowButton';
import WindowControls from '../app/components/WindowControls';

describe('<WindowControls />', () => {
  it('renders with window buttons', () => {
    const wrapper = shallow(<WindowControls />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.containsMatchingElement(<WindowButton />)).toBe(true);
  });
});

describe('<WindowButton />', () => {
  const color = '#fff';
  const icon = 'icon';

  it('fires onClick func', () => {
    let testVal = null;
    const onClick = () => {
      testVal = 'clicked';
    };
    const wrapper = shallow(
      <WindowButton color={color} icon={icon} onClick={onClick} />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.simulate('click');
    expect(testVal).toBe('clicked');
  });

  it('displays correct icon', () => {
    const wrapper = shallow(
      <WindowButton color={color} icon={icon} onClick={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper
        .find(FontAwesome)
        .at(0)
        .prop('name')
    ).toBe(icon);
  });
});
