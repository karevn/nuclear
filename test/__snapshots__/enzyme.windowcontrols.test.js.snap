// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WindowButton /> displays correct icon 1`] = `
<div
  className="window_button_container"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
    }
  }
>
  <FontAwesome
    name="icon"
  />
</div>
`;

exports[`<WindowButton /> fires onClick func 1`] = `
<div
  className="window_button_container"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
    }
  }
>
  <FontAwesome
    name="icon"
  />
</div>
`;

exports[`<WindowControls /> renders with window buttons 1`] = `
<div
  className="window_controls_container"
>
  <WindowButton
    icon="window-minimize"
    onClick={[Function]}
  />
  <WindowButton
    icon="window-maximize"
    onClick={[Function]}
  />
  <WindowButton
    icon="close"
    onClick={[Function]}
  />
</div>
`;
